diff --git tensorflow/dtensor/BUILD tensorflow/dtensor/BUILD
index b8117e927a2..b79f991107b 100644
--- tensorflow/dtensor/BUILD
+++ tensorflow/dtensor/BUILD
@@ -24,6 +24,6 @@ package_group(
         "//tensorflow/core/...",
         "//tensorflow/dtensor/...",
         "//tensorflow/python/...",
-        "//tensorflow_federated/...",  #  Needs access to C++ API for dtensor device
+        "public",  #  Needs access to C++ API for dtensor device
     ],
 )
