diff --git tensorflow/BUILD tensorflow/BUILD
index b7199470078..1ccd3788f3d 100644
--- tensorflow/BUILD
+++ tensorflow/BUILD
@@ -1005,8 +1005,16 @@
 # Instead, please use public APIs or public build rules TF provides.
 # If you need functionality that is not exposed, we will work with you to expand our public APIs.
 # TODO(b/173549186): Move Google-internal TF code out of learning/brain
+
+# BEGIN: TensorFlow Federated modifications
 package_group(
     name = "internal",
+    packages = ["//..."],
+)
+# END: TensorFlow Federated modifications
+
+package_group(
+    name = "unused_internal",
     packages = [
         "//devtools/python/indexer/...",
         "//learning/brain/keras/...",
