diff --git tensorflow/BUILD tensorflow/BUILD
index b5b076847b6..5f9938d42f2 100644
--- tensorflow/BUILD
+++ tensorflow/BUILD
@@ -1045,7 +1045,7 @@ package_group(
         "//smartass/brain/configure/...",
         "//tensorflow/...",
         "//tensorflow_decision_forests/...",
-        "//tensorflow_federated/...",
+        "public",
         "//third_party/cloud_tpu/convergence_tools/sdc_monitoring/...",
         "//third_party/cloud_tpu/inference_converter/...",
         "//third_party/py/cloud_ml_autoflow/...",
