diff --git a/tensorflow/BUILD b/tensorflow/BUILD
index b58696813e7..d840aa12a4c 100644
--- a/tensorflow/BUILD
+++ b/tensorflow/BUILD
@@ -884,6 +884,7 @@ selects.config_setting_group(
 package_group(
     name = "internal",
     packages = [
+        "public",
         "//devtools/python/indexer/...",
         "//learning/brain/keras/...",
         "//learning/brain/mlir/...",
@@ -897,7 +898,6 @@ package_group(
         "//smartass/brain/...",
         "//tensorflow/...",
         "//tensorflow_decision_forests/...",
-        "//tensorflow_federated/...",
         "//tensorflow_models/google/recml/...",
         "//third_party/cloud_tpu/convergence_tools/sdc_monitoring/...",
         "//third_party/cloud_tpu/inference_converter/...",
