diff --git tensorflow/BUILD tensorflow/BUILD
index 92e11a68e8d..096e4ce0b66 100644
--- tensorflow/BUILD
+++ tensorflow/BUILD
@@ -797,17 +797,7 @@ config_setting(
 package_group(
     name = "internal",
     packages = [
-        "//learning/brain/keras/...",
-        "//learning/brain/mlir/...",
-        "//learning/brain/tfrt/...",
-        "//learning/lib/ami/simple_ml/...",
-        "//tensorflow/...",
-        "//tensorflow_decision_forests/...",
-        "//tensorflow_federated/...",
-        "//third_party/cloud_tpu/inference_converter/...",
-        "//third_party/py/envlogger/...",
-        "//third_party/py/keras/...",
-        "//third_party/yggdrasil_decision_forests/...",
+        "//...",
     ],
 )

