diff --git tensorflow/BUILD tensorflow/BUILD
index 202553cd531..171eb04665c 100644
--- tensorflow/BUILD
+++ tensorflow/BUILD
@@ -1039,7 +1039,7 @@ package_group(
         "//smartass/brain/configure/...",
         "//tensorflow/...",
         "//tensorflow_decision_forests/...",
-        "//tensorflow_federated/...",
+        "public",
         "//third_party/cloud_tpu/convergence_tools/sdc_monitoring/...",
         "//third_party/cloud_tpu/inference_converter/...",
         "//third_party/py/cloud_ml_autoflow/...",
