diff --git tensorflow/BUILD tensorflow/BUILD
index cf97153fe06..504736cd5d6 100644
--- tensorflow/BUILD
+++ tensorflow/BUILD
@@ -897,19 +897,7 @@ config_setting(
 package_group(
     name = "internal",
     packages = [
-        "//devtools/python/indexer/...",
-        "//learning/brain/keras/...",
-        "//learning/brain/mlir/...",
-        "//learning/brain/tfrt/...",
-        "//learning/lib/ami/simple_ml/...",
-        "//tensorflow/...",
-        "//tensorflow_decision_forests/...",
-        "//tensorflow_federated/...",
-        "//third_party/cloud_tpu/inference_converter/...",
-        "//third_party/py/cloud_ml_autoflow/...",
-        "//third_party/py/envlogger/...",
-        "//third_party/py/keras/...",
-        "//third_party/yggdrasil_decision_forests/...",
+        "//...",
     ],
 )

