diff --git a/tensorflow/core/framework/tensor_testutil.cc b/tensorflow/core/framework/tensor_testutil.cc
index 4718013093b..1d54f8d122b 100644
--- a/tensorflow/core/framework/tensor_testutil.cc
+++ b/tensorflow/core/framework/tensor_testutil.cc
@@ -16,6 +16,7 @@ limitations under the License.
 #include "tensorflow/core/framework/tensor_testutil.h"
 
 #include <cmath>
+#include <iomanip>
 
 #include "tensorflow/core/framework/tensor.h"
 #include "tensorflow/core/platform/types.h"
