diff --git tensorflow/compiler/tf2xla/BUILD tensorflow/compiler/tf2xla/BUILD
index 436ea8653ed..0a73258d391 100644
--- tensorflow/compiler/tf2xla/BUILD
+++ tensorflow/compiler/tf2xla/BUILD
@@ -38,10 +38,20 @@
     ],
 )
 
+# BEGIN: TensorFlow Federated modification
 package_group(
     name = "friends",
     includes = [":internal"],
     packages = [
+        "//...",
+    ],
+)
+# END: TensorFlow Federated modification
+
+package_group(
+    name = "unused_friends",
+    includes = [":internal"],
+    packages = [
         "//tensorflow/...",
         "//tensorflow_federated/cc/core/impl/executors/...",
         "//tensorflow_models/...",
