diff --git tensorflow/compiler/tf2xla/BUILD tensorflow/compiler/tf2xla/BUILD
index d82a2c17bc1..9f299e3ac69 100644
--- tensorflow/compiler/tf2xla/BUILD
+++ tensorflow/compiler/tf2xla/BUILD
@@ -47,14 +47,7 @@ package_group(
     name = "friends",
     includes = [":internal"],
     packages = [
-        "//tensorflow/...",
-        "//tensorflow_federated/cc/core/impl/executors/...",
-        "//tensorflow_models/...",
-        "//third_party/deepmind/deepmind_research/density_functional_approximation_dm21/...",
-        "//third_party/mlir_edge/model_curriculum/iree/...",
-        "//third_party/mlperf/submissions/training/v0_7/models/...",
-        "//third_party/py/keras/...",
-        "//waymo/ml/deploy/benchmark/...",
+        "//...",
     ],
 )

