diff --git tensorflow/compiler/tf2xla/BUILD tensorflow/compiler/tf2xla/BUILD
index ef32bdb19a8..7eac4b4ab75 100644
--- tensorflow/compiler/tf2xla/BUILD
+++ tensorflow/compiler/tf2xla/BUILD
@@ -46,7 +46,7 @@ package_group(
     packages = [
         "//platforms/performance/automl/...",
         "//tensorflow/...",
-        "//tensorflow_federated/cc/core/impl/executors/...",
+        "public",
         "//tensorflow_models/...",
         "//third_party/deepmind/deepmind_research/density_functional_approximation_dm21/...",
         "//third_party/mlir_edge/model_curriculum/iree/...",
