diff --git federated_language/proto/BUILD federated_language/proto/BUILD
--- federated_language/proto/BUILD
+++ federated_language/proto/BUILD
@@ -15,4 +15,3 @@
-load("@protobuf//bazel:cc_proto_library.bzl", "cc_proto_library")
-load("@protobuf//bazel:proto_library.bzl", "proto_library")
-load("@protobuf//bazel:py_proto_library.bzl", "py_proto_library")
+load("@com_github_grpc_grpc//bazel:python_rules.bzl", "py_proto_library")
+load("@rules_cc//cc:defs.bzl", "cc_proto_library")
 load("@rules_python//python:defs.bzl", "py_library")
