diff --git federated_language/common_libs/BUILD federated_language/common_libs/BUILD
index b16a6df..5d5378f 100644
--- federated_language/common_libs/BUILD
+++ federated_language/common_libs/BUILD
@@ -37,7 +37,6 @@ py_library(
     name = "golden",
     testonly = True,
     srcs = ["golden.py"],
-    deps = ["@federated_language_pypi//absl_py"],
 )
 
 py_test(
@@ -57,26 +56,19 @@ py_test(
         "golden_test_goldens/test_check_string_succeeds.expected",
         "golden_test_goldens/test_check_string_updates.expected",
     ],
-    deps = [
-        ":golden",
-        "@federated_language_pypi//absl_py",
-    ],
+    deps = [":golden"],
 )
 
 py_library(
     name = "py_typecheck",
     srcs = ["py_typecheck.py"],
-    deps = ["@federated_language_pypi//typing_extensions"],
 )
 
 py_test(
     name = "py_typecheck_test",
     size = "small",
     srcs = ["py_typecheck_test.py"],
-    deps = [
-        ":py_typecheck",
-        "@federated_language_pypi//absl_py",
-    ],
+    deps = [":py_typecheck"],
 )
 
 py_library(
@@ -88,10 +80,7 @@ py_test(
     name = "retrying_test",
     size = "small",
     srcs = ["retrying_test.py"],
-    deps = [
-        ":retrying",
-        "@federated_language_pypi//absl_py",
-    ],
+    deps = [":retrying"],
 )
 
 py_library(
@@ -102,36 +91,24 @@ py_library(
 py_library(
     name = "structure",
     srcs = ["structure.py"],
-    deps = [
-        ":py_typecheck",
-        "@federated_language_pypi//attrs",
-        "@federated_language_pypi//dm_tree",
-    ],
+    deps = [":py_typecheck"],
 )
 
 py_test(
     name = "structure_test",
     size = "small",
     srcs = ["structure_test.py"],
-    deps = [
-        ":structure",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//attrs",
-    ],
+    deps = [":structure"],
 )
 
 py_library(
     name = "tracing",
     srcs = ["tracing.py"],
-    deps = ["@federated_language_pypi//absl_py"],
 )
 
 py_test(
     name = "tracing_test",
     size = "small",
     srcs = ["tracing_test.py"],
-    deps = [
-        ":tracing",
-        "@federated_language_pypi//absl_py",
-    ],
+    deps = [":tracing"],
 )
diff --git federated_language/compiler/BUILD federated_language/compiler/BUILD
index f4b7a0e..98ec3bc 100644
--- federated_language/compiler/BUILD
+++ federated_language/compiler/BUILD
@@ -41,8 +41,6 @@ py_library(
         "//federated_language/proto:array_py_pb2",
         "//federated_language/types:array_shape",
         "//federated_language/types:dtype_utils",
-        "@federated_language_pypi//ml_dtypes",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -53,9 +51,6 @@ py_test(
         ":array",
         "//federated_language/proto:array_py_pb2",
         "//federated_language/proto:data_type_py_pb2",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//ml_dtypes",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -135,8 +130,6 @@ py_test(
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
         "//federated_language/types:type_analysis",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -153,7 +146,6 @@ py_library(
         "//federated_language/types:placements",
         "//federated_language/types:type_analysis",
         "//federated_language/types:typed_object",
-        "@federated_language_pypi//numpy",
         "@protobuf//:protobuf_python",
     ],
 )
@@ -174,10 +166,6 @@ py_test(
         "//federated_language/proto:data_type_py_pb2",
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//dm_tree",
-        "@federated_language_pypi//ml_dtypes",
-        "@federated_language_pypi//numpy",
         "@protobuf//:protobuf_python",
     ],
 )
@@ -200,8 +188,6 @@ py_test(
         "//federated_language/proto:computation_py_pb2",
         "//federated_language/types:computation_types",
         "//federated_language/types:type_factory",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -212,7 +198,6 @@ py_library(
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
         "//federated_language/types:type_factory",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -220,10 +205,7 @@ py_test(
     name = "intrinsic_defs_test",
     size = "small",
     srcs = ["intrinsic_defs_test.py"],
-    deps = [
-        ":intrinsic_defs",
-        "@federated_language_pypi//absl_py",
-    ],
+    deps = [":intrinsic_defs"],
 )
 
 py_library(
@@ -242,8 +224,6 @@ py_test(
         ":transformation_utils",
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
         "@protobuf//:protobuf_python",
     ],
 )
@@ -272,7 +252,5 @@ py_test(
         ":tree_analysis",
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
diff --git federated_language/computation/BUILD federated_language/computation/BUILD
index 168f6e8..559109b 100644
--- federated_language/computation/BUILD
+++ federated_language/computation/BUILD
@@ -77,8 +77,6 @@ py_test(
         "//federated_language/context_stack:context_stack_impl",
         "//federated_language/proto:computation_py_pb2",
         "//federated_language/types:computation_types",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -107,8 +105,6 @@ py_test(
         "//federated_language/context_stack:context_stack_impl",
         "//federated_language/proto:computation_py_pb2",
         "//federated_language/types:computation_types",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -131,8 +127,6 @@ py_test(
         ":function_utils",
         "//federated_language/common_libs:structure",
         "//federated_language/types:computation_types",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -157,7 +151,5 @@ py_test(
         "//federated_language/proto:computation_py_pb2",
         "//federated_language/types:computation_types",
         "//federated_language/types:type_conversions",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
diff --git federated_language/context_stack/BUILD federated_language/context_stack/BUILD
index baac0d1..a294aaf 100644
--- federated_language/context_stack/BUILD
+++ federated_language/context_stack/BUILD
@@ -55,7 +55,6 @@ py_test(
     deps = [
         ":context",
         ":context_stack_impl",
-        "@federated_language_pypi//absl_py",
     ],
 )
 
@@ -75,7 +74,6 @@ py_test(
         ":context",
         ":context_stack_impl",
         ":context_stack_test_utils",
-        "@federated_language_pypi//absl_py",
     ],
 )
 
diff --git federated_language/executor/BUILD federated_language/executor/BUILD
index 4a099af..9938844 100644
--- federated_language/executor/BUILD
+++ federated_language/executor/BUILD
@@ -48,7 +48,6 @@ py_library(
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
         "//federated_language/types:type_conversions",
-        "@federated_language_pypi//dm_tree",
     ],
 )
 
@@ -59,7 +58,6 @@ py_test(
     deps = [
         ":async_execution_context",
         "//federated_language/executor:executor_base",
-        "@federated_language_pypi//absl_py",
     ],
 )
 
@@ -80,8 +78,6 @@ py_test(
         "//federated_language/common_libs:structure",
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
diff --git federated_language/federated_context/BUILD federated_language/federated_context/BUILD
index 8d90095..59dbb00 100644
--- federated_language/federated_context/BUILD
+++ federated_language/federated_context/BUILD
@@ -57,8 +57,6 @@ py_test(
         "//federated_language/context_stack:context_stack_impl",
         "//federated_language/context_stack:runtime_error_context",
         "//federated_language/types:computation_types",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -89,8 +87,6 @@ py_test(
         "//federated_language/context_stack:context_stack_impl",
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -118,8 +114,6 @@ py_test(
         "//federated_language/computation:function_utils",
         "//federated_language/context_stack:context_stack_impl",
         "//federated_language/types:computation_types",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -141,7 +135,6 @@ py_library(
         "//federated_language/types:placements",
         "//federated_language/types:type_analysis",
         "//federated_language/types:type_factory",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -158,8 +151,6 @@ py_test(
         "//federated_language/context_stack:context_stack_test_utils",
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -182,7 +173,6 @@ py_library(
         "//federated_language/types:placements",
         "//federated_language/types:type_conversions",
         "//federated_language/types:typed_object",
-        "@federated_language_pypi//attrs",
     ],
 )
 
@@ -200,9 +190,6 @@ py_test(
         "//federated_language/context_stack:context_stack_impl",
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//attrs",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -232,7 +219,5 @@ py_test(
         "//federated_language/context_stack:context_stack_impl",
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
diff --git federated_language/program/BUILD federated_language/program/BUILD
index b2e8ba9..bcc6c02 100644
--- federated_language/program/BUILD
+++ federated_language/program/BUILD
@@ -73,8 +73,6 @@ py_test(
         "//federated_language/context_stack:context_stack_impl",
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -84,7 +82,6 @@ py_library(
     deps = [
         ":release_manager",
         ":value_reference",
-        "@federated_language_pypi//absl_py",
     ],
 )
 
@@ -94,9 +91,6 @@ py_test(
     deps = [
         ":logging_release_manager",
         ":program_test_utils",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//dm_tree",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -115,9 +109,6 @@ py_test(
     deps = [
         ":memory_release_manager",
         ":program_test_utils",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//dm_tree",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -134,7 +125,6 @@ py_library(
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
         "//federated_language/types:type_conversions",
-        "@federated_language_pypi//dm_tree",
     ],
 )
 
@@ -151,9 +141,6 @@ py_test(
         "//federated_language/federated_context:federated_computation",
         "//federated_language/types:computation_types",
         "//federated_language/types:placements",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//dm_tree",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -170,10 +157,7 @@ py_library(
 py_test(
     name = "program_state_manager_test",
     srcs = ["program_state_manager_test.py"],
-    deps = [
-        ":program_state_manager",
-        "@federated_language_pypi//absl_py",
-    ],
+    deps = [":program_state_manager"],
 )
 
 py_library(
@@ -185,9 +169,6 @@ py_library(
         "//federated_language/common_libs:py_typecheck",
         "//federated_language/common_libs:serializable",
         "//federated_language/types:computation_types",
-        "@federated_language_pypi//attrs",
-        "@federated_language_pypi//dm_tree",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -198,8 +179,6 @@ py_library(
         ":structure_utils",
         ":value_reference",
         "//federated_language/common_libs:py_typecheck",
-        "@federated_language_pypi//attrs",
-        "@federated_language_pypi//dm_tree",
     ],
 )
 
@@ -209,9 +188,6 @@ py_test(
     deps = [
         ":program_test_utils",
         ":release_manager",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//dm_tree",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -232,19 +208,13 @@ py_test(
         ":program_test_utils",
         ":serialization_utils",
         "//federated_language/types:computation_types",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
 py_library(
     name = "structure_utils",
     srcs = ["structure_utils.py"],
-    deps = [
-        "//federated_language/common_libs:py_typecheck",
-        "@federated_language_pypi//attrs",
-        "@federated_language_pypi//dm_tree",
-    ],
+    deps = ["//federated_language/common_libs:py_typecheck"],
 )
 
 py_test(
@@ -253,9 +223,6 @@ py_test(
     deps = [
         ":program_test_utils",
         ":structure_utils",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//dm_tree",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -266,7 +233,6 @@ py_library(
         ":structure_utils",
         "//federated_language/types:computation_types",
         "//federated_language/types:typed_object",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -276,8 +242,5 @@ py_test(
     deps = [
         ":program_test_utils",
         ":value_reference",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//dm_tree",
-        "@federated_language_pypi//numpy",
     ],
 )
diff --git federated_language/types/BUILD federated_language/types/BUILD
index 0c62a63..7c1b6c2 100644
--- federated_language/types/BUILD
+++ federated_language/types/BUILD
@@ -50,7 +50,6 @@ py_test(
     deps = [
         ":array_shape",
         "//federated_language/proto:array_py_pb2",
-        "@federated_language_pypi//absl_py",
     ],
 )
 
@@ -65,9 +64,6 @@ py_library(
         "//federated_language/common_libs:structure",
         "//federated_language/proto:array_py_pb2",
         "//federated_language/proto:computation_py_pb2",
-        "@federated_language_pypi//attrs",
-        "@federated_language_pypi//numpy",
-        "@federated_language_pypi//typing_extensions",
     ],
 )
 
@@ -81,31 +77,19 @@ py_test(
         "//federated_language/common_libs:structure",
         "//federated_language/proto:computation_py_pb2",
         "//federated_language/proto:data_type_py_pb2",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//attrs",
-        "@federated_language_pypi//numpy",
     ],
 )
 
 py_library(
     name = "dtype_utils",
     srcs = ["dtype_utils.py"],
-    deps = [
-        "//federated_language/proto:data_type_py_pb2",
-        "@federated_language_pypi//ml_dtypes",
-        "@federated_language_pypi//numpy",
-    ],
+    deps = ["//federated_language/proto:data_type_py_pb2"],
 )
 
 py_test(
     name = "dtype_utils_test",
     srcs = ["dtype_utils_test.py"],
-    deps = [
-        ":dtype_utils",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//ml_dtypes",
-        "@federated_language_pypi//numpy",
-    ],
+    deps = [":dtype_utils"],
 )
 
 py_library(
@@ -117,10 +101,7 @@ py_test(
     name = "placements_test",
     size = "small",
     srcs = ["placements_test.py"],
-    deps = [
-        ":placements",
-        "@federated_language_pypi//absl_py",
-    ],
+    deps = [":placements"],
 )
 
 py_library(
@@ -130,8 +111,6 @@ py_library(
         ":array_shape",
         ":computation_types",
         ":placements",
-        "@federated_language_pypi//ml_dtypes",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -143,9 +122,6 @@ py_test(
         ":computation_types",
         ":placements",
         ":type_analysis",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//ml_dtypes",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -158,9 +134,6 @@ py_library(
         ":typed_object",
         "//federated_language/common_libs:py_typecheck",
         "//federated_language/common_libs:structure",
-        "@federated_language_pypi//attrs",
-        "@federated_language_pypi//dm_tree",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -174,9 +147,6 @@ py_test(
         ":type_conversions",
         ":typed_object",
         "//federated_language/common_libs:structure",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//attrs",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -193,8 +163,6 @@ py_test(
     deps = [
         ":computation_types",
         ":type_factory",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
@@ -212,8 +180,6 @@ py_test(
         ":computation_types",
         ":placements",
         ":type_transformations",
-        "@federated_language_pypi//absl_py",
-        "@federated_language_pypi//numpy",
     ],
 )
 
